src/middleware/router.js

```diff
@@ -537,6 +556,110 @@ function sendHttpRequest (ctx, route, options) {
   })
 }
 
+// Send secondary route request
+const sendSecondaryRouteHttpRequest = (ctx, route, options) => {
+  return new Promise((resolve, reject) => {
+    const response = {}
+    let { downstream } = ctx.state
+    let method = http
+
+    if (route.secured) {
+      method = https
+    }
+
+    const routeReq = method.request(options)
```

Comment:
Is it possible to refactor the methods for sending out the stream to be re-usable. Alot of this code seems to be duplicated on the sending of the primary request